internal/provider/resource_group.go

```diff
@@ -118,19 +123,15 @@ func resourceGroupCreate(ctx context.Context, d *schema.ResourceData, meta inter
 		}
 		gcsmr, err := grps.SetMembers(ctx, gcr.Item.Id, gcr.Item.Version, memberIds)
 		if err != nil {
-			return diag.Errorf("error setting principals on role: %v", err)
+			return diag.Errorf("error setting members on group: %v", err)
 		}
 		if gcsmr == nil {
 			return diag.Errorf("group nil after setting members")
 		}
-		raw = gcsmr.GetResponse().Map
+		apiResponse = gcsmr.GetResponse().Map
 	}
 
-	if err := setFromGroupResponseMap(d, raw); err != nil {
-		return diag.FromErr(err)
-	}
-
-	return nil
+	return errs
```

Comment:
You can even return `nil` here and keep this diff even smaller. It will assign `nil` to the `errs` variable, then call the deferred function, which may or may not mutate the value before returning.

Reply:
I did a test locally and noticed that if I return nil at the end of the function, then only the value appended in the defer function is returned. With the current code I think its fine if we want to do that, but in the future if someone appends a warning level diagnostic to the slice and they do not return errs explicitly, then the warning value will not be in the slice.

Reply:
I see the concern. When I read this I was confused that we were returning `errs` here since it's untouched by the path to this code. It is unusual to return a value that we know is `nil`. I'd prefer to make this `nil`, and as you say, refactor to return `errs` if we do start assigning to it in this function.